pageserver/src/tenant/timeline.rs

```diff
@@ -4734,11 +4732,11 @@ fn is_send() {
 
 /// Add a suffix to a layer file's name: .{num}.old
 /// Uses the first available num (starts at 0)
-fn rename_to_backup(path: &Path) -> anyhow::Result<()> {
+fn rename_to_backup(path: &Utf8Path) -> anyhow::Result<()> {
     let filename = path
         .file_name()
-        .ok_or_else(|| anyhow!("Path {} don't have a file name", path.display()))?
-        .to_string_lossy();
+        .ok_or_else(|| anyhow!("Path {path} don't have a file name",))?
+        .to_string();
```

Comment:
```suggestion
        .ok_or_else(|| anyhow!("Path {path} don't have a file name",))?;
```

If I understand correctly, this would be perfectly fine with `let filename` being a `&str`.